manager/client/client.go

```diff
@@ -793,6 +795,254 @@ WaitForAllData:
+		if err != nil {
+			return aprs, err
+		}
+		reqs = append(reqs, req)
+
+		req, err = hc.createTaskRequest(ctx, nv, account, bl.Height, shared.ReqIDGetReward)
+		if err != nil {
+			return aprs, err
+		}
+		reqs = append(reqs, req)
+
+		if prevDayEndHeight != 0 {
+			rows = append(rows, dataRow{
+				dayStart:    dayStart,
+				startHeight: prevDayEndHeight,
+				endHeight:   bl.Height,
+			})
+		}
+
+		if isBefore {
+			break
+		}
+
+		dayStart = dayEnd
+		dayEnd = nextDayEnd
+
+		prevDayEndHeight = bl.Height
+	}
+
+	hc.logger.Info("[Client] Sending request data:", zap.Any("requests", reqs))
+	var respAwait *structs.Await
+	if len(reqs) > 0 {
+		respAwait, err = hc.sender.Send(reqs)
+		if err != nil {
+			hc.logger.Error("[Client] Error sending data", zap.Error(err))
+			return aprs, fmt.Errorf("error sending data in GetRewardApr: %w", err)
+		}
+
+		defer respAwait.Close()
+	}
+
+	buff := &bytes.Buffer{}
+	dec := json.NewDecoder(buff)
+
+	balancesMap := make(map[uint64][]shared.TransactionAmount)
+	rewardsMap := make(map[uint64][]shared.TransactionAmount)
+
+	accountBalanceCounter := responseTypes.WithLabels(nv.Network, nv.ChainID, "GetRewardApr", "AccountBalance")
+	rewardCounter := responseTypes.WithLabels(nv.Network, nv.ChainID, "GetRewardApr", "Reward")
+
+	var receivedFinals int
+WaitForAllData:
+	for {
+		if len(reqs) == 0 {
+			break
+		}
+
+		select {
+		case <-ctx.Done():
+			return aprs, errors.New("request timed out")
+		case response := <-respAwait.Resp:
+			hc.logger.Debug("[Client] Get account balance  received data:", zap.Any("type", response.Type), zap.Any("response", response))
+
+			if response.Error.Msg != "" {
+				return aprs, fmt.Errorf("error getting response: %s", response.Error.Msg)
+			}
+
+			if response.Type == "AccountBalance" {
+				buff.Reset()
+				buff.ReadFrom(bytes.NewReader(response.Payload))
+				b := &shared.GetAccountBalanceResponse{}
+				err := dec.Decode(b)
+				if err != nil {
+					return aprs, fmt.Errorf("error decoding account balance: %w", err)
+				}
+				balancesMap[b.Height] = b.Balances
+				accountBalanceCounter.Inc()
+			}
+			if response.Type == "Reward" {
+				buff.Reset()
+				buff.ReadFrom(bytes.NewReader(response.Payload))
+				b := &shared.GetRewardResponse{}
+				err := dec.Decode(b)
+				if err != nil {
+					return aprs, fmt.Errorf("error decoding reward: %w", err)
+				}
+				rewardsMap[b.Height] = b.Rewards
+				rewardCounter.Inc()
+			}
+
+			if response.Final {
+				receivedFinals++
+			}
+
+			if receivedFinals == len(reqs) {
+				hc.logger.Info("[Client] Received All GetRewardApr", zap.Any("network", nv.Network), zap.Any("chain_id", nv.ChainID), zap.Int("account_balance_count", len(balancesMap)), zap.Int("reward_count", len(rewardsMap)))
+				break WaitForAllData
+			}
+		}
+	}
+
+	rewardTxTypes := []string{"delegate", "withdraw_delegator_reward", "begin_unbonding", "begin_redelegate"}
```

Comment:
there was a bug in rewards call, need to add missing type:
```suggestion
	rewardTxTypes := []string{"delegate", "undelegate", "withdraw_delegator_reward", "begin_unbonding", "begin_redelegate"}
```